pulp_container/app/serializers.py

```diff
@@ -505,6 +508,38 @@ class ManifestCopySerializer(CopySerializer):
     )
 
 
+class RemoveImageSerializer(serializers.Serializer):
+    """
+    A serializer for parsing and validating data associated with the image removal.
+    """
+
+    digest = serializers.CharField(help_text="sha256 of the Manifest file")
+
+    def validate(self, data):
+        """
+        Validate and extract the latest repository version, manifest, and tags from the passed data.
+        """
+        new_data = {}
+        new_data.update(self.initial_data)
```

Comment:
What is happening here?
Should this not be `new_data = super().validate(data)`?
Does this circumvent the field validations?
Why do we refer to a repository here, and there is no repository field?

Reply:
There is no need to call the base `validate()` method because it does literally nothing. The field for a repository is initialized automatically. We already use such a technique here: https://github.com/pulp/pulp_container/blob/8274e1ed1b83ccfe2c7eec1294582dfb9d8a71ea/pulp_container/app/serializers.py#L324-L335

It does not circumvent the validation because of this:
https://github.com/pulp/pulp_container/pull/247/files#diff-31c44b2812e50f3171248db027f439e8e64248cb84067fc7c305de74cdb9a86fR875-R880